### PR TITLE
add pyupgrade (#2253)

### PR DESCRIPTION
**Pull request by lielfr**
## Description

Added https://github.com/asottile/pyupgrade to .pre-commit-config.yam (#2253)
Tested it with `pre-commit run --all-files`, and also added the changes of running the hook in a separate commit.
I had to add the `--keep-runtime-typing` flag, otherwise it would break the tests.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #2253 Add pyupgrade to pre-commit

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation. - **I'm pretty sure it's not**.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes. - **Is it relevant here?**
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage). - **Ran tests and also tested the example app.py specified in the README. Everything seems to work fine.**

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Liel Fridman for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by lielfr** [comment]
Also, I'm not sure if the release should be patch (as it is now) or minor. Would happily change it if needed

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Release type: patch
+
+
+Added pyupgrade to pre-commit hooks. Also ran it and made sure there are no (unexpected) test breakages after the changes.
```
Comment (estyxx) [inline RELEASE.md:4]: I will just write:

This release adds pyupgrade to pre-commit

**Review by estyxx — COMMENTED** [review]
Thank you!
Think patch is fine!

File: RELEASE.md (hunk 1 above)
Comment (lielfr) [reply RELEASE.md]: done, thanks!

**Review by estyxx — APPROVED** [review]
LGTM

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Release type: patch
+
+
+This release adds pyupgrade to pre-commit
```
Comment (patrick91) [inline RELEASE.md:4]:
```suggestion
This release adds pyupgrade to our CI and includes some minor changes to keep our codebase modern.
```

**Review by patrick91 — APPROVED** [review]
Thank you!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**